### PR TITLE
[quant][onnx] Add support to convert max_pool2d quantized pytorch op to C2

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34629 [quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2
* **#33945 [quant][onnx] Add support to convert max_pool2d quantized pytorch op to C2**

Summary:
Add mapping for this operator in symbolics

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_max_pool2d

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20433681](https://our.internmc.facebook.com/intern/diff/D20433681)

### CODE REVIEW

File: torch/onnx/symbolic_caffe2.py [hunk 1]
```diff
@@ -154,6 +154,24 @@ def upsample_nearest2d(g, input, output_size, align_corners=None, scales_h=None,
...
+        return max_pool2d(g, input, kernel_size, stride, padding, dilation, ceil_mode)
+    kwargs = {
+        "strides_i": stride,
+        "pads_i": padding + padding,
+        "kernel_i": kernel_size[0],
+        "order_s": "NHWC",
+        "Y_scale_f": input.node()["Y_scale"],
+        "Y_zero_point_i": input.node()["Y_zero_point"],
+    }
+    input = nchw2nhwc(g, input)
```
Comment (z-a-f) [inline torch/onnx/symbolic_caffe2.py]: Is there a way to check if it is already in NHWC? That way we could add a fast pass here

**Review by z-a-f — APPROVED** [review]
LGTM

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 2]
```diff
@@ -204,5 +204,19 @@ def forward(self, x, y):
...
+                super(QMaxPool2dModule, self).__init__()
+                self.quant1 = torch.quantization.QuantStub()
+                self.dequant = torch.quantization.DeQuantStub()
+
+            def forward(self, x):
+                res = torch.nn.functional.max_pool2d(self.quant1(x), kernel_size=2, stride=1, padding=0)
+                return self.dequant(res)
+
+        x = np.random.rand(1, 2, 8, 8).astype("float32")
+        self.generic_test(QMaxPool2dModule(), (x,), input_names=["x"])
```
Comment (raghuramank100) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py]: Can we check for an exact match here for maxpool?  Also, should we test for different values of stride/padding?

File: torch/onnx/symbolic_caffe2.py (hunk 1 above)
Comment (supriyar) [reply torch/onnx/symbolic_caffe2.py]: The way this conversion flow is written, it always adds these permutes before and after conv ops. Else branches in the graph become hard to handle. Once we convert the model we can use a script to remove the unnecessary permutes from the graph.

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py (hunk 2 above)
Comment (houseroad) [reply test/onnx/test_pytorch_onnx_caffe2_quantized.py]: If we want to check for exact match for maxpool, then I think we need to add parameter decimal to generic_test

**Review by houseroad — COMMENTED** [review]
Overall, looks good.

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 3]
```diff
@@ -204,5 +204,19 @@ def forward(self, x, y):
...
+                super(QMaxPool2dModule, self).__init__()
+                self.quant1 = torch.quantization.QuantStub()
+                self.dequant = torch.quantization.DeQuantStub()
+
+            def forward(self, x):
+                res = torch.nn.functional.max_pool2d(self.quant1(x), kernel_size=2, stride=1, padding=0)
+                return self.dequant(res)
+
+        x = np.random.rand(1, 2, 8, 8).astype("float32")
+        self.generic_test(QMaxPool2dModule(), (x,), input_names=["x"], decimal=0)
```
Comment (raghuramank100) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py]: We should be using a larger value for decimal, from the docs:
https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.testing.assert_almost_equal.html
abs(desired-actual) < 1.5 * 10**(-decimal)

Can we use decimal of 5?

**Review by raghuramank100 — APPROVED** [review]
